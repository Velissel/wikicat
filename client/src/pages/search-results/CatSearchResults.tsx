import { isEmpty } from 'lodash';
import React from 'react';
import { Alert, Col, Row } from 'react-bootstrap';
import CatCard from 'src/components/CatCard';
import CatSpinner from 'src/components/CatSpinner';

interface CatSearchResultsProps {
  error?: string;
  cats: any[];
  loading: boolean;
  loaded: boolean;
}
export default function CatSearchResults(props: CatSearchResultsProps) {
  const { error, loading, loaded, cats } = props;

  if (error) {
    return <Alert variant="danger">{error}</Alert>;
  }

  if (loading) {
    return <CatSpinner />;
  }

  if (!loaded) {
    return <>Start search cat breeds by putting keyworkds in</>;
  }

  if (isEmpty(cats)) {
    return <Alert variant="warning">Nothing is found</Alert>;
  }

  return (
    <Row data-testid="cat-search-results">
      {cats.map((cat) => {
        return (
          <Col key={cat.id} xs={12} sm={6} md={4} lg={3} className="mb-3">
            <CatCard cat={cat} />
          </Col>
        );
      })}
    </Row>
  );
}
