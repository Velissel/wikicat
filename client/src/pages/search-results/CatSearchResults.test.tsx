import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CatSearchResults from './CatSearchResults';

describe('CatSearchResults component', () => {
  it('should show error', () => {
    const { container } = render(
      <MemoryRouter>
        <CatSearchResults
          cats={[]}
          loading={false}
          loaded={false}
          error="ERROR"
        />
      </MemoryRouter>,
    );

    expect(screen.getByText('ERROR')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should show loading', () => {
    const { container } = render(
      <MemoryRouter>
        <CatSearchResults cats={[]} loading={true} loaded={false} />
      </MemoryRouter>,
    );

    expect(screen.getByText('Loading...')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should show prmot user to start search', () => {
    const { container } = render(
      <MemoryRouter>
        <CatSearchResults cats={[]} loading={false} loaded={false} />
      </MemoryRouter>,
    );

    expect(
      screen.getByText('Start search cat breeds by putting keyworkds in'),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should show no cats message', () => {
    const { container } = render(
      <MemoryRouter>
        <CatSearchResults cats={[]} loading={false} loaded={true} />
      </MemoryRouter>,
    );

    expect(screen.getByText('Nothing is found')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should show cat list', () => {
    const cat = {
      id: 'id',
      image: 'url',
      name: 'name',
      description: 'description',
    };
    const { container } = render(
      <MemoryRouter>
        <CatSearchResults cats={[cat]} loading={false} loaded={true} />
      </MemoryRouter>,
    );

    expect(screen.getByTestId(`cat-card-${cat.id}`)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
