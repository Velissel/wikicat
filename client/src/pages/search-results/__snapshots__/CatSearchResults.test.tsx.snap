// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatSearchResults component should show cat list 1`] = `
<div>
  <div
    class="row"
    data-testid="cat-search-results"
  >
    <div
      class="mb-3 col-lg-3 col-md-4 col-sm-6 col-12"
    >
      <div
        class="card"
        data-testid="cat-card-id"
      >
        <img
          class="card-img-top image"
          data-testid="cat-img"
          src="url"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            name
          </div>
          <p
            class="description card-text"
          >
            description
          </p>
          <a
            href="/kitty/id"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              View Breed Details
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CatSearchResults component should show error 1`] = `
<div>
  <div
    class="fade alert alert-danger show"
    role="alert"
  >
    ERROR
  </div>
</div>
`;

exports[`CatSearchResults component should show loading 1`] = `
<div>
  <div
    class="spinner"
  >
    <div
      class="spinner-header"
    >
      <img
        alt="logo"
        class="spinner-logo"
        src="logo.svg"
      />
      <p>
        Loading...
      </p>
    </div>
  </div>
</div>
`;

exports[`CatSearchResults component should show no cats message 1`] = `
<div>
  <div
    class="fade alert alert-warning show"
    role="alert"
  >
    Nothing is found
  </div>
</div>
`;

exports[`CatSearchResults component should show prmot user to start search 1`] = `
<div>
  Start search cat breeds by putting keyworkds in
</div>
`;
